NVIDIA-Merlin/Merlin
# Add example notebook for training and inference on Sagemaker

This PR adds an example notebook that demonstrates how to use Merlin container for training and inference on the AWS Sagemaker platform. It can be roughly split into three sections:
1. Test the training script locally to make sure that the script runs as expected,
2. Train the model on Sagemaker by calling `sagemaker.Estimator.fit(entry_point='train.py')`, and
3. Create an endpoint with Triton, and test it by sending a request.

Note:
- The notebook builds a custom image (built from `merlin-tensorflow` image) and uploads it to ECR. This step can be simplified/removed by installing the `sagemaker-training` library directly in the Merlin image, so I opened a separate PR: https://github.com/NVIDIA-Merlin/Merlin/pull/707.

Steps for testing (NV-specific):
- Ask Ben to give you access to the AWS account. After he adds you to the group, wait a couple of hours for permissions to propagate.
- If permissions have propagated, you will see `AWS NVIDIA Account` on https://myapps.microsoft.com/. This can be used to access the console.
- Install [aws-azure-login](https://github.com/DTN-Public/aws-azure-login) and run `aws-azure-login --mode=gui` to authenticate. This will create the credientials in `$HOME/.aws`. More details here: https://gitlab-master.nvidia.com/kaizen-efficiency/wiki/-/wikis/Guides/aws-cli-sso
- As described in [README](https://github.com/NVIDIA-Merlin/Merlin/pull/692/files#diff-d1479b6bb0466c56f2b4db0d8acf2724c2acc62f1aaf90826337088c6740efb7R23-R42), add `-v $HOME/.aws:/root/.aws` to your usual docker run command.
